app/services/idv/steps/doc_auth_base_step.rb

```diff
@@ -27,13 +27,17 @@ def idv_failure(result)
             step_name: self.class.name,
             remaining_attempts: throttle.remaining_count,
           )
-          redirect_to idv_session_errors_exception_url
+          redirect_to idv_session_errors_exception_url(
+            from: request.original_url,
+          )
```

Comment:
I was sorta hoping this could be a simple token like "doc_auth" vs. "in_person", though I guess that's not readily available without a bit of extra effort. I wonder if we'd have any issues like [described in this StackOverflow comment](https://stackoverflow.com/a/31171250) using `original_url` ?

Alternatively, there are some ways we could get a token from e.g. `self.class.name` or `@flow.class.name` or `@flow.controller.class.name`, but if we did that, we might want to clean it up so that it's not a namespaced Ruby class name.

Reply:
Another thought could be to implement an abstract method or variable in `VerifyBaseStep`, implement that for each flow, and pass that through to or reference from `idv_failure`.

```rb
module Idv
  module Steps
    class DocAuthBaseStep < Flow::BaseStep
      def idv_failure
        # ...
        redirect_to exception_url
      end

      def exception_url
        idv_session_errors_exception_url
      end
    end
  end
end

module Idv
  module Steps
    module Ipp
      class VerifyStep < VerifyBaseStep
        def exception_url
          idv_session_errors_exception_url(flow: :in_person)
        end
      end
    end
  end
end

module Idv
  module Steps
    class VerifyStep < VerifyBaseStep
      def exception_url
        idv_session_errors_exception_url(flow: :doc_auth)
      end
    end
  end
end

```

Reply:
I think this'd be easier to determine if there was an intermediate step in the class hierarchy for IPP, but seeing as there are shared steps I decided to reduce this to the path instead of the full URL to reduce the probability of the issue mentioned on SO.